### PR TITLE
Add PortabilityService EndPoint to ApiPortService

### PR DESCRIPTION
**Pull request by JJVertical**

### CODE REVIEW

File: src/Microsoft.Fx.Portability/ApiPortService.cs [hunk 1]
```diff
@@ -102,6 +102,19 @@ public ApiPortService(string endpoint, ProductInformation info)
             return await _client.CallAsync<IReadOnlyCollection<ApiDefinition>>(HttpMethod.Get, url.Url);
         }
 
+        /// <summary>
+        /// Calls the PortabilityService DocIds endpoint and return the result
+        /// </summary>
+        /// <param name="docIds">Enumerable of DocIds</param>
+        public async Task<ServiceResponse<IReadOnlyCollection<ApiInformation>>> GetDocIds(IEnumerable<string> docIds)
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability/ApiPortService.cs]: Can you add this to IApiPortService

File: src/Microsoft.Fx.Portability/ApiPortService.cs [hunk 2]
```diff
@@ -102,6 +102,19 @@ public ApiPortService(string endpoint, ProductInformation info)
...
+        {
+            var url = UrlBuilder.Create(Endpoints.FxApi);
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability/ApiPortService.cs]: Do you need to use UrlBuilder here?

File: tests/Microsoft.Fx.Portability.Tests/ApiPortServiceTests.cs [hunk 3]
```diff
@@ -15,5 +24,45 @@ public void VerifyParameterChecks()
             Assert.Throws<ArgumentOutOfRangeException>(() => new ApiPortService(string.Empty, new ProductInformation("")));
             Assert.Throws<ArgumentOutOfRangeException>(() => new ApiPortService(" \t", new ProductInformation("")));
         }
+
+        [Fact(Skip = SkipReason)]
+        public void ApiPortService_GetDocIdsWithValidDocId()
+        {
+            var apiPortService = new ApiPortService(
```
Comment (twsouthwick) [inline tests/Microsoft.Fx.Portability.Tests/ApiPortServiceTests.cs]: These tests should be rewritten with an HttpClient mock - create an overload of ApiPortService that takes an HttpMessageHandler and then you can mock it fairly easily.

File: tests/Microsoft.Fx.Portability.Tests/ApiPortServiceTests.cs [hunk 4]
```diff
@@ -2,12 +2,21 @@
...
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace Microsoft.Fx.Portability.Tests
 {
     public class ApiPortServiceTests
     {
+        private const string ServiceEndpoint = "http://portability.cloudapp.net";
```
Comment (twsouthwick) [inline tests/Microsoft.Fx.Portability.Tests/ApiPortServiceTests.cs]: The tests should never rely on the service

File: tests/Microsoft.Fx.Portability.Tests/ApiPortServiceTests.cs [hunk 5]
```diff
@@ -15,5 +24,45 @@ public void VerifyParameterChecks()
...
+                "M:System.Xml.Serialization.XmlSerializer.Serialize(System.Xml.XmlWriter,System.Object)",
+                "M:System.Xml.Serialization.XmlSerializer.Serialize(System.IO.Stream,System.Object,System.Xml.Serialization.XmlSerializerNamespaces)",
+                "M:System.Xml.Serialization.XmlSerializer.Serialize(System.IO.TextWriter,System.Object,System.Xml.Serialization.XmlSerializerNamespaces)",
+                "M:System.Xml.Serialization.XmlSerializer.Serialize(System.Xml.XmlWriter,System.Object,System.Xml.Serialization.XmlSerializerNamespaces)",
+                "M:System.Xml.Serialization.XmlSerializer.Serialize(System.Xml.XmlWriter,System.Object,System.Xml.Serialization.XmlSerializerNamespaces,System.String)"
+            };
+
+            var result = apiPortService.GetDocIds(docIds).Result.Response;
+            Assert.Equal(docIds.Count(), result.Count());
+            Assert.Equal(0, docIds.Except(result.Select(r => r.Definition.DocId)).Count());
```
Comment (conniey) [inline tests/Microsoft.Fx.Portability.Tests/ApiPortServiceTests.cs]: I forgot about using that LINQ trick to check that two collections are equal. :+1:

File: src/Microsoft.Fx.Portability.Offline/OfflineApiPortService.cs [hunk 6]
```diff
@@ -116,5 +116,10 @@ public Task<ServiceResponse<ApiInformation>> GetApiInformationAsync(string docId
 
             return new ServiceResponse<IReadOnlyCollection<ApiDefinition>>(result);
         }
+
+        public Task<ServiceResponse<IReadOnlyCollection<ApiInformation>>> QueryDocIdsAsync(IEnumerable<string> docIds)
+        {
+            throw new NotImplementedException();
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability.Offline/OfflineApiPortService.cs]: Why did you not implement this? This is needed for offline support

File: src/Microsoft.Fx.Portability/Properties/version.txt [hunk 7]
```diff
@@ -1 +1 @@
-1.3.1
+1.3.2
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability/Properties/version.txt]: I would bump this to 1.4.0 since it's a new API, not a bug fix

**Comment by twsouthwick** [comment]
LGTM

**Pull request merged**